src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,264 @@
+        try {
+            FileWriter fw = new FileWriter(SAVE_PATH);
+            fw.write(storeStr);
+            fw.close();
+        } catch (IOException e) {
+            System.err.println("Failed to write to save file!" + e.getMessage());
+        }
+    }
+
+    private static void loadTaskList() {
+        String saveStr = "";
+        try {
+            FileReader fr = new FileReader(SAVE_PATH);
+            int i;
+            while ((i=fr.read()) != -1){
+                saveStr += ((char)i);
+            }
+            System.out.println(saveStr);
+            fr.close();
+        } catch (IOException e){
+            System.err.println("Failed to open save file!" + e.getMessage());
+            return;
+        }
+        // ATTEMPTING PARSE
+
+        JSONArray ja = (JSONArray) JSONValue.parse(saveStr);
+//        Object[] ja = ((JSONArray) JSONValue.parse(saveStr)).toArray();
+//        for (int i = 0; i < ja.length; i++){
+//            Object curItem = ja[i];
+//            toDoList.add(new Task())
+//        }
+        for(Object o: ja){
+            if ( o instanceof JSONObject) {
+                JSONObject jo = (JSONObject)o;
+                String type = (String)jo.get("type");
+                System.out.println("type: " + type);
+                Task toAdd = null;
+                if (type.equals("TODO")) {
+                    toAdd = new ToDo((String)jo.get("description"));
+                } else if (type.equals("DEADLINE")) {
+                    toAdd = new Deadline((String)jo.get("description"),(String)jo.get("time"));
+                } else {
+                    toAdd = new Event((String)jo.get("description"),(String)jo.get("time"));
+                }
+                toDoList.add(toAdd);
+            }
+        }
+
+    }
+
+    private static Integer getTodoListSize() {
+        return toDoList.size();
+    }
+
+    private static void listToDo(List<Task> tdList) {
+        System.out.println("Here are the tasks in your list:");
+        int idx = 1;
+        for (Task todo : tdList) {
+            System.out.println(String.format("%d. %s", idx, todo.getStatus()));
+            idx += 1;
+        }
+    }
+
+    public static void printHello() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello, I'm \n" + logo);
+        System.out.println("What can I do for you?\n");
+    }
+
+    public static void printBye() {
+        System.out.println("Bye. Hope to see you again soon!\n");
+    }
+
+    public static void updateMark(String isMark, String[] cmd) {
+        int idx;
+        try {
+            idx = Integer.parseInt(cmd[1]) - 1;
+        } catch (NumberFormatException e) {
+            System.out.println("OOPS, param provided is NOT an integer");
+            return;
+        }
```

Comment:
You have two try catch blocks in this method. To make the [happy path more prominent](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent), try using a guard clause for handling invalid or non-integer indices.